packages/core-data/src/actions.js

```diff
@@ -155,11 +161,21 @@ export function receiveEmbedPreview( url, preview ) {
 export function* deleteEntityRecord( kind, name, recordId, query ) {
 	const entities = yield getKindEntities( kind );
 	const entity = find( entities, { kind, name } );
-	let error;
-	let deletedRecord = false;
 	if ( ! entity ) {
 		return;
 	}
+	const lock = yield acquireStoreLock( [ kind, name, recordId ], true );
+	try {
+		return yield* doDeleteEntityRecord( entity, recordId, query );
+	} finally {
+		yield releaseStoreLock( lock );
+	}
+}
+
+function* doDeleteEntityRecord( entity, recordId, query ) {
```

Comment:
A function name beginning with "do" is a bit smelly to me. Functions, by definition, are things that "do" stuff 😛

Just an idea: Maybe we could remove this boiler plate with a higher order function similar to `createSelector`?

```js
export const deleteEntityRecord = createAtomicAction(
	function* ( kind, name, recordId, query ) {
		// meat goes here
	},
	( kind, name, recordId ) => [ kind, name, recordId ]
);
```

Reply:
Good idea!

Reply:
This ended up being more complex than simple so I abandoned the idea for now. Let's revisit if/when needed!